### PR TITLE
Add rec score

### PR DESCRIPTION
**Pull request by tink2123**
* add rec score in infer_rec
* fix export model for attention
* add error message
* add drop last args
* support variable shape for chinese inference

### CODE REVIEW

File: configs/rec/rec_chinese_lite_train.yml [hunk 1]
```diff
@@ -9,15 +9,18 @@ Global:
   eval_batch_step: 2000
   train_batch_size_per_card: 256
   test_batch_size_per_card: 256
+  drop_last: false
```
Comment (dyning) [inline configs/rec/rec_chinese_lite_train.yml:12]: drop_last参数是不是不添加的好？

File: doc/recognition.md [hunk 2]
```diff
@@ -184,7 +184,7 @@ python3 tools/eval.py -c configs/rec/rec_chinese_lite_train.yml -o Global.checkp
 
 ```
 # 预测英文结果
-python3 tools/infer_rec.py -c configs/rec/rec_chinese_lite_train.yml -o Global.checkpoints={path/to/weights}/best_accuracy TestReader.infer_img=doc/imgs_words/en/word_1.jpg
+python3 tools/infer_rec.py -c configs/rec/rec_chinese_lite_train.yml -o Global.checkpoints={path/to/weights}/best_accuracy Global.infer_img=doc/imgs_words/en/word_1.png
```
Comment (dyning) [inline doc/recognition.md:187]: rec_chinese_lite_train,中文模型为什么识别en？

File: ppocr/data/det/dataset_traversal.py [hunk 3]
```diff
@@ -61,8 +62,9 @@ def batch_iter_reader():
                 if len(batch_outs) == self.batch_size:
                     yield batch_outs
                     batch_outs = []
-            if len(batch_outs) != 0:
-                yield batch_outs
+            if not self.drop_last:
```
Comment (dyning) [inline ppocr/data/det/dataset_traversal.py:65]: 训练的时候drop

File: ppocr/data/rec/dataset_traversal.py [hunk 4]
```diff
@@ -165,34 +180,42 @@ def __init__(self, params):
         self.loss_type = params['loss_type']
         self.max_text_length = params['max_text_length']
         self.mode = params['mode']
+        self.infer_img = params['infer_img']
+        self.tps = False
```
Comment (dyning) [inline ppocr/data/rec/dataset_traversal.py:184]: 将tps改为use_tps

File: ppocr/data/rec/img_tools.py [hunk 5]
```diff
@@ -66,8 +92,17 @@ def process_image(img,
...
-                  max_text_length=None):
-    norm_img = resize_norm_img(img, image_shape)
+                  max_text_length=None,
+                  tps=None,
+                  infer_mode=False):
+    if not infer_mode or char_ops.character_type == "en":
+        norm_img = resize_norm_img(img, image_shape)
+    else:
+        if tps != None and char_ops.character_type == "ch":
+            image_shape = [3, 32, 320]
```
Comment (dyning) [inline ppocr/data/rec/img_tools.py:102]: 这块有问题，应该和conf里面一致

File: tools/infer/predict_rec.py [hunk 6]
```diff
@@ -80,26 +84,43 @@ def __call__(self, img_list):
...
-                beg = predict_lod[rno]
-                end = predict_lod[rno + 1]
-                probs = predict_batch[beg:end, :]
-                ind = np.argmax(probs, axis=1)
-                blank = probs.shape[1]
-                valid_ind = np.where(ind != (blank - 1))[0]
-                score = np.mean(probs[valid_ind, ind[valid_ind]])
-                rec_res.append([preds_text, score])
+
+            if self.rec_algorithm != "RARE":
```
Comment (dyning) [inline tools/infer/predict_rec.py:88]: 通过loss判断

File: configs/rec/rec_icdar15_train.yml [hunk 7]
```diff
@@ -9,14 +9,17 @@ Global:
   eval_batch_step: 500
   train_batch_size_per_card: 256
   test_batch_size_per_card: 256
+  drop_last: false
   image_shape: [3, 32, 100]
   max_text_length: 25
   character_type: en
   loss_type: ctc
   reader_yml: ./configs/rec/rec_icdar15_reader.yml
-  pretrain_weights: ./pretrain_models/rec_mv3_none_bilstm_ctc/best_accuracy
```
Comment (dyning) [inline configs/rec/rec_icdar15_train.yml:17]: 删除？

File: doc/inference.md [hunk 8]
```diff
@@ -165,6 +165,12 @@ STAR-Net文本识别模型推理，可以执行如下命令：
 ```
 python3 tools/infer/predict_rec.py --image_dir="./doc/imgs_words_en/word_336.png" --rec_model_dir="./inference/starnet/" --rec_image_shape="3, 32, 100" --rec_char_type="en"
 ```
+
+RARE 文本识别模型推理，可以执行如下命令：
```
Comment (dyning) [inline doc/inference.md:169]: 基于Attention损失的识别模型推理

File: configs/rec/rec_chinese_lite_train.yml (hunk 1 above)
Comment (tink2123) [reply configs/rec/rec_chinese_lite_train.yml]: done

File: ppocr/data/det/dataset_traversal.py (hunk 3 above)
Comment (tink2123) [reply ppocr/data/det/dataset_traversal.py:65]: done

File: doc/recognition.md (hunk 2 above)
Comment (tink2123) [reply doc/recognition.md]: rec_chinese同时支持中英文

File: ppocr/data/rec/dataset_traversal.py (hunk 4 above)
Comment (tink2123) [reply ppocr/data/rec/dataset_traversal.py]: done

File: ppocr/data/rec/img_tools.py (hunk 5 above)
Comment (tink2123) [reply ppocr/data/rec/img_tools.py]: done

File: tools/infer/predict_rec.py (hunk 6 above)
Comment (tink2123) [reply tools/infer/predict_rec.py]: done

File: configs/rec/rec_icdar15_train.yml (hunk 7 above)
Comment (tink2123) [reply configs/rec/rec_icdar15_train.yml:17]: done

File: doc/inference.md (hunk 8 above)
Comment (tink2123) [reply doc/inference.md:173]: done

File: ppocr/data/rec/dataset_traversal.py [hunk 9]
```diff
@@ -100,13 +104,18 @@ def __call__(self, process_id):
             process_id = 0
 
         def sample_iter_reader():
-            if self.mode == 'test':
+            if self.mode != 'train' and self.infer_img is not None:
```
[the remark this answers is not in the document]
Comment (tink2123) [reply ppocr/data/rec/dataset_traversal.py:108]: done

File: README.md [hunk 10]
```diff
@@ -36,6 +36,8 @@ PaddleOCR旨在打造一套丰富、领先、且实用的OCR工具库，助力
 
 #### 2.inference模型下载
 
+*windows 环境下没有如果没有安装wget,下载模型时可将链接复制到浏览器中下载，并解压放置在相应目录下*
```
Comment (dyning) [inline README.md:39]: 语句不通

File: configs/rec/rec_icdar15_train.yml [hunk 11]
```diff
@@ -1,22 +1,24 @@
 Global:
   algorithm: CRNN
-  use_gpu: true
+  use_gpu: false
```
Comment (dyning) [inline configs/rec/rec_icdar15_train.yml:3]: 为什么默认设为False呢？

**Comment by SixGodGG** [comment]
add rec score in infer_rec   这个score  和之前的那个置信度（默认0.5）有什么区别？

File: README.md (hunk 10 above)
Comment (tink2123) [reply README.md]: done thx

File: configs/rec/rec_icdar15_train.yml (hunk 11 above)
Comment (tink2123) [reply configs/rec/rec_icdar15_train.yml]: fixed,thx

**Comment by tink2123** [comment]
> add rec score in infer_rec 这个score 和之前的那个置信度（默认0.5）有什么区别？

本质没有区别，使用infer_rec 预测文本时，会将每个预测结果的得分都打印出来，不会过滤
